magento/pwa-studio
# [PWA-767] [Mini-cart] Guest user can't add product to cart if cart is expired.

## Bug Description

Steps -
1. Add product to cart as guest.
2. Wait for guest cart to expire (No sure whats the time, need to check)
3. Try adding product again.

*Expected - *
Toast should trigger informing cart expiration (ensure messaging is user friendly).
New Cart should be created.
Product should get added to new cart.

## Solution
The use case of someone leaving their tab open for the default span of thirty days is a bit out there, and it would be best to treat this exactly like any of our other cart initialization recovery. I've instead just added this expired cart message to cart recovery on initial page load, and removed the logic that was popping a Toast on cart query in the new MiniCart component.

This now should just silently recover, and the user should be able to add items to their cart.

## Related Issue





* [[PWA-767](https://jira.corp.magento.com/browse/PWA-767)] [Mini-cart] Guest user can't add product to cart if cart is expired.

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


The easiest way I found to test this is with database access, making this a little difficult 😬 
1. Add products to your cart
2. `SELECT quote_id FROM quote_id_mask WHERE masked_id = '<cart-id>'`
3. `UPDATE quote SET is_active = 0 WHERE entity_id = <quote-id-from-above>`
4. Simulate customer coming back after > 30 days by just reloading
5. Verify your old cart gets wiped and new cart provisioned by adding a item to your cart

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
